pootle/apps/pootle_store/models.py

```diff
@@ -1437,8 +1439,11 @@ def delete(self, *args, **kwargs):
         super(Store, self).delete(*args, **kwargs)
 
         self.clear_all_cache(parents=False, children=False)
-        if parent is not None:
-            parent.update_all_cache()
+        for p in parents:
+            p.update_all_cache()
+
+            for vfcti in vfctis:
```

Comment:
http://www.slideshare.net/pirhilton/how-to-name-things-the-hardest-problem-in-programming